maxfw/core/utils.py

```diff
@@ -0,0 +1,55 @@
+from flask import abort
+from maxfw.utils.image_utils import ImageProcessor
+
+
+def redirect_errors_to_flask(func):
+    """
+    This decorator function will capture all Pythonic errors and return them as flask errors.
+
+    If you are looking to disable this functionality, please remove this decorator from the `apply_transforms()` module
+    under the ImageProcessor class.
+    """
+
+    def inner(*args, **kwargs):
```

Comment:
Is this still a WIP for catching all the relevant errors / error messages?

Reply:
@kmh4321 do you want to finish this or should I take this up?

Reply:
@MLnick I wanted to check with you on this, are we supposed to look for ALL possible exceptions and map them to 3-4 flask (user-friendly) exceptions? or should we just do these 3 mentioned here ie ValueError, TypeError and a bare Exception?

Reply:
The idea here is to handle the case of invalid user input. We can distinguish between invalid input and an application error.

For invalid user input, we should return a `400` response, together with a message that informs the user what was wrong with the input, so they can fix it and send valid input.

The main cases:
- user sends non-image input. Message should be something like "Provided input should be a valid image (JPG, PNG, ..." (as we have done in most models to date). Most of the time we will handle various image types and we do the conversions required internally (e.g. to 3-channel RGB from RGBA input, or converting to grayscale etc).
- user sends image input that is bad somehow - this is likely a corrupted image? Or perhaps a strange format that may hit some of the edge cases with respect to num channels etc (perhaps a 1- or 2-channel saved image that we're trying to convert to RGB). In this case we want a message that is as specific as possible, but without exposing internals (e.g. PIL specific stuff).

(I may have missed some other valid cases here)

The other cases are actually not user errors - it will be because the transform pipeline is not set up correctly (e.g. incorrect use of a `Standardize` component etc). This should _not_ return a `400` but a `500`. If we just let the exception get thrown, Flask will return a 500 response without the stack trace (this is logged).

Reply:
So I guess what I'm saying here is - we want to catch `TypeError` and perhaps certain cases of `ValueError` (and not even necessarily all instances of these errors, some will be for invalid input and some will be for app errors).

We want to avoid generically catching all exceptions and returning `400`s for all cases.

Reply:
I agree, however most (if not all) the exceptions we are catching are written by us (none of them expose PIL exceptions). 
Most of the exceptions which will be caught here are defined in this file and they are all written by us to be as generic as possible so even if I return it as-is, it shouldn't be an issue.

However, based on above points you made, I am catching 2 specific exceptions (from the 30-odd exceptions we have):
- when the `convert_to_pil()` function fails because of an invalid input such as a text file/non image file. 
- when the `convert_to_pil()` function fails because of incorrect dimensions inputted. For example if some arbitrary numpy array of size (10,10,10,10,10) was inputted. 

These are the only errors I could see which would be correlating to an invalid user input. All others are due to incorrect configuration of the pipeline and/or invalid arguments being passed to different stages of the pipeline and so while there are exceptions being thrown for these, we don't have any flask aborts for them so flask would automatically return `500 -  Internal Server Error` for these. However I do feel that these are not really "server" side issues because the root cause is invalid user config/usage.

Reply:
I've accordingly added exceptions in d879144.

Reply:
LGTM